packages/contracts/contracts/LenderCommitmentForwarder/extensions/FlashRolloverLoan.sol

```diff
@@ -0,0 +1,370 @@
+
+contract FlashRolloverLoan is
+    ICommitmentRolloverLoan,
+    IFlashLoanSimpleReceiver
+{
+    using AddressUpgradeable for address;
+    using NumbersLib for uint256;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ITellerV2 public immutable TELLER_V2;
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ILenderCommitmentForwarder public immutable LENDER_COMMITMENT_FORWARDER;
+
+    address public immutable POOL_ADDRESSES_PROVIDER;
+
+    event RolloverLoanComplete(
+        address borrower,
+        uint256 originalLoanId,
+        uint256 newLoanId,
+        uint256 fundsRemaining
+    );
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _tellerV2,
+        address _lenderCommitmentForwarder,
+        address _poolAddressesProvider
+    ) {
+        TELLER_V2 = ITellerV2(_tellerV2);
+        LENDER_COMMITMENT_FORWARDER = ILenderCommitmentForwarder(
+            _lenderCommitmentForwarder
+        );
+        POOL_ADDRESSES_PROVIDER = _poolAddressesProvider;
+    }
+
+    modifier onlyFlashLoanPool() {
+        require(
+            msg.sender == address(POOL()),
+            "FlashRolloverLoan: Must be called by FlashLoanPool"
+        );
+
+        _;
+    }
+
+    /*
+    need to pass loanId and borrower 
+    */
+
+    struct RolloverCallbackArgs {
+        uint256 loanId;
+        address borrower;
+        uint256 borrowerAmount;
+        bytes acceptCommitmentArgs;
+    }
+
+    /**
+     * @notice Allows a borrower to rollover a loan to a new commitment.
+     * @param _loanId The bid id for the loan to repay
+     * @param _flashLoanAmount The amount to flash borrow.
+     * @param _acceptCommitmentArgs Arguments for the commitment to accept.
+     * @return newLoanId_ The ID of the new loan created by accepting the commitment.
+     */
+
+    /*
+ 
+The flash loan amount can naively be the exact amount needed to repay the old loan 
+
+If the new loan pays out (after fees) MORE than the  aave loan amount+ fee) then borrower amount can be zero 
+
+ 1) I could solve for what the new loans payout (before fees and after fees) would NEED to be to make borrower amount 0...
+
+*/
+
+    function rolloverLoanWithFlash(
+        uint256 _loanId,
+        uint256 _flashLoanAmount,
+        uint256 _borrowerAmount, //an additional amount borrower may have to add
+        AcceptCommitmentArgs calldata _acceptCommitmentArgs
+    ) external returns (uint256 newLoanId_) {
+        address borrower = TELLER_V2.getLoanBorrower(_loanId);
+        require(borrower == msg.sender, "CommitmentRolloverLoan: not borrower");
+
+        // Get lending token and balance before
+        address lendingToken = TELLER_V2.getLoanLendingToken(_loanId);
+
+        if (_borrowerAmount > 0) {
+            IERC20(lendingToken).transferFrom(
+                borrower,
+                address(this),
+                _borrowerAmount
+            );
+        }
```

Comment:
do we even need this if we can just flash loan the amount?

Reply:
yes we do need this because the borrower may need to pay a difference. For example if the new loan is smaller than the old loan in principal